src/migrate/index.js

```diff
@@ -108,9 +115,49 @@ export default class Migrator {
     });
   }
 
+  _createMigrationLockTable(tableName) {
+    return this.knex.schema.createTable(tableName, function(t) {
+      t.boolean('is_locked');
+    });
+  }
+
+  _getLockTableName(tableName) {
+    return tableName + '_lock';
+  }
+
+  _isLocked() {
+    var tableName = this._getLockTableName(this.config.tableName);
+    return this.knex(tableName)
+      .count('is_locked as count')
+      .then(function(data) {
+        if (data[0].count > 0) {
+          return true;
+        }
+        return false;
+      });
+  }
+
+  _lockMigrations() {
```

Comment:
It is locked while it is updated, but reading has already been done before updating, which causes the problems. 
I mean this kind of scenario:
```
Process 1 read lock row: locked: false
Process 2 read lock row: locked: false
Process 1 write lock => locked : true
Process 2 write lock => locked : true
Process 1 start running migrations
Process 2 start running migrations
```
If you make transaction before reading the lock and say that when I select this row I'm going to update it later on in this same transaction, it applies read lock for that row and it is not released until transaction ends. Scenario above will change to:
```
Process 1 start transaction
Process 2 start transaction
Process 1 select read lock row for update: locked: false // select for update causes read lock for row
Process 2 select read lock row for update... // row has already been selected for update in other transaction... must wait
Process 1 write lock => locked : true
Process 1 end transaction
Process 2 read lock row: locked: true // row's read lock was freed... we can continue now
Process 1 start running migrations
Process 2 exit
Process 1 set lock = false
Process 1 exit
```